front/src/applications/operationalStudies/helpers/formatTrainScheduleSummaries.ts

```diff
@@ -0,0 +1,84 @@
+import dayjs from 'dayjs';
+import { compact, keyBy } from 'lodash';
+
+import type {
+  LightRollingStockWithLiveries,
+  SimulationSummaryResult,
+  TrainScheduleResult,
+} from 'common/api/osrdEditoastApi';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/TimetableV2/types';
+import { formatToIsoDate, isoDateToMs } from 'utils/date';
+import { jouleToKwh } from 'utils/physics';
+import { formatKmValue } from 'utils/strings';
+import { ISO8601Duration2sec } from 'utils/timeManipulation';
+import type { Dictionary } from 'utils/types';
+
+const getRelevantTrainSchedules = (
+  rawTrainSchedules: Dictionary<TrainScheduleResult>,
+  trainIds: number[]
+) => {
+  const relevantTrainSchedules: TrainScheduleResult[] = [];
+
+  trainIds.forEach((trainId) => {
+    const trainSchedule = rawTrainSchedules[trainId];
+    if (trainSchedule) relevantTrainSchedules.push(trainSchedule);
+  });
+
+  return relevantTrainSchedules;
+};
+
+const formatTrainScheduleSummaries = (
+  trainIds: number[],
+  rawSummaries: Dictionary<SimulationSummaryResult>,
+  rawTrainSchedules: Dictionary<TrainScheduleResult>,
+  rollingStocks: LightRollingStockWithLiveries[]
+): Dictionary<TrainScheduleWithDetails> => {
+  const relevantTrainSchedules = getRelevantTrainSchedules(rawTrainSchedules, trainIds);
```

Comment:
```suggestion
  const relevantTrainSchedules = trainIds.map(trainId => rawTrainSchedules[trainId]).filter(Boolean)
```
Since the logic is now more concise, the `getRelevantTrainSchedules` function may no longer be needed.

Reply:
Yes you're right !